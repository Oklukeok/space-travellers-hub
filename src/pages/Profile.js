import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';
// import { fetchMissions } from '../redux/missions/missions';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  /* const missions = useSelector((state) => state.missions.missions); */
  const dispatch = useDispatch();
  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(fetchRockets());
    }
  });

  /*  useEffect(()=> {
    if(missions.length === 0) {
      dispatch(fetchMissions());
    }
  }); */

  return (
    <div className="Profile">

      {/* <div className="reserved-missions-wrapper">
        <h3 className="profile-title profile-title-mission">
          My Missions
        </h3>
        <ul className="reserved-profile reserved-profile-mission ">
          {
          missions.map((mission) => (
            mission.reserved ? (
              <li key={mission.mission_id} className="reserved-mission-item">
                {mission.mission_name}
              </li>
            ) : ''
          ))
          }

        </ul>
      </div> */}

      <div className="reserved-rockets-wrapper">
        <h3 className="profile-title profile-title-rocket">
          My Rockets
        </h3>
        <ul className="reserved-profile reserved-profile-rocket ">
          {
          rockets.map((rocket) => (
            rocket.reserved ? (
              <li key={rocket.rocket_id} className="reserved-rocket-item">
                {rocket.rocket_name}
              </li>
            ) : ''
          ))
          }

        </ul>
      </div>

    </div>
  );
};

export default Profile;
