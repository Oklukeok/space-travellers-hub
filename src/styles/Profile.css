.profile {
  display: flex;
  align-items: center;
  padding: 0 120px 0 200px;
}

.reserved-rockets-wrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
}

.reserved-profile {
  list-style: none;
  border-top: 1px solid #999;
  border-right: 1px solid #999;
  border-left: 1px solid #999;
  width: 100%;
}

.reserved-item {
  border-bottom: 1px solid #999;
  padding: 7px 0 20px 10px;
}

/* .reserved-item:last-child {
  border-bottom: none;
} */
