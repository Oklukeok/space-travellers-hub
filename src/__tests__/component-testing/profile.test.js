import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Profile from '../../pages/Profile';
import store from '../../redux/configureStore';

test('Render My profile', () => {
  const renderProfile = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Profile />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(renderProfile).toMatchSnapshot();
});
