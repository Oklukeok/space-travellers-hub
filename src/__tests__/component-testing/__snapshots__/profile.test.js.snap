// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render My profile 1`] = `
<div
  className="profile"
>
  <div
    className="reserved-rockets-wrapper"
  >
    <h3
      className="profile-title profile-title-rocket"
    >
      My Rockets
    </h3>
    <div
      className="reserved-profile reserved-profile-rocket "
    />
  </div>
</div>
`;

exports[`Render Navbar 1`] = `
Array [
  <div
    className="profile"
  >
    <div
      className="reserved-rockets-wrapper"
    >
      <h3
        className="profile-title profile-title-rocket"
      >
        My Rockets
      </h3>
      <div
        className="reserved-profile reserved-profile-rocket "
      />
    </div>
  </div>,
  ")",
]
`;

exports[`Testing if Profile component renders correctly 1`] = `
<div
  className="profile"
>
  <div
    className="reserved-rockets-wrapper"
  >
    <h3
      className="profile-title profile-title-rocket"
    >
      My Rockets
    </h3>
    <div
      className="reserved-profile reserved-profile-rocket "
    />
  </div>
</div>
`;
